contracts/rewards/src/contract/execute.rs

```diff
@@ -156,18 +204,229 @@ mod test {
+        participation.insert(Addr::unchecked("worker_2"), 5);
+        participation.insert(Addr::unchecked("worker_3"), 7);
+        let event_count = 10;
+        let mut cur_height = block_height_started;
+        for i in 0..event_count {
+            for (w, p) in &participation {
+                if i < *p {
+                    contract
+                        .record_participation(
+                            i.to_string().try_into().unwrap(),
+                            w.clone(),
+                            worker_contract.clone(),
+                            cur_height,
+                        )
+                        .unwrap();
+                }
+            }
+            cur_height = cur_height + 1;
+        }
+
+        {
+            let tally_store = tally_store.read().unwrap();
+            let tally = tally_store
+                .get(&(
+                    worker_contract.clone(),
+                    contract
+                        .get_current_epoch(block_height_started)
+                        .unwrap()
+                        .epoch_num,
+                ))
+                .unwrap();
+            assert_eq!(tally.event_count, event_count);
+            assert_eq!(tally.participation.len(), participation.len());
+            for (w, p) in participation {
+                assert_eq!(tally.participation.get(&w), Some(&p));
+            }
+        }
+    }
+
+    /// Tests that the participation event is recorded correctly when the event spans multiple epochs
+    #[test]
+    fn record_participation_epoch_boundary() {
+        let cur_epoch_num = 1u64;
+        let block_height_started = 250u64;
+        let epoch_duration = 100u64;
+        let event_store = Arc::new(RwLock::new(HashMap::new()));
+        let tally_store = Arc::new(RwLock::new(HashMap::new()));
+
+        let mut contract = setup_with_stores(
+            cur_epoch_num,
+            block_height_started,
+            epoch_duration,
+            event_store.clone(),
+            tally_store.clone(),
+        );
+
+        let worker_contract = Addr::unchecked("some contract");
+
+        let workers = vec![
+            Addr::unchecked("worker_1"),
+            Addr::unchecked("worker_2"),
+            Addr::unchecked("worker_3"),
+        ];
+        let mut cur_height = block_height_started + epoch_duration - 1;
+        let prev_epoch = contract.get_current_epoch(cur_height).unwrap();
+        for w in &workers {
+            contract
+                .record_participation(
+                    "some event".to_string().try_into().unwrap(),
+                    w.clone(),
+                    worker_contract.clone(),
+                    cur_height,
+                )
+                .unwrap();
+            cur_height = cur_height + 1;
+        }
+        let cur_epoch = contract.get_current_epoch(cur_height).unwrap();
+        assert_ne!(prev_epoch.epoch_num, cur_epoch.epoch_num);
```

Comment:
```suggestion
        assert_eq!(starting_epoch_num + 1, cur_epoch.epoch_num);
```
rename `cur_epoch_num` above to `starting_epoch_num`